### ISSUE TITLE
rust: VmDock seems not implementing function execution in named module

### ISSUE THREAD

**Issue by yorkie** [WasmEdge/WasmEdge#2371]
## Description

See https://github.com/WasmEdge/WasmEdge/blob/master/bindings/rust/wasmedge-sdk/src/dock.rs#L35, the comment describes the function `run_func`:

> Runs an exported WASM function registered in a named or active module.

However the implementation seems to use `active_module()` only, is this a bug or any reason that WasmEdge not going to support this feature for named module?

**Comment by apepkuss** [comment]
@yorkie The motivation for introducing VmDock was to provide the same capabilities of [wasmedge-bindgen](https://github.com/second-state/wasmedge-bindgen) in WasmEdge Rust SDK. So, VmDock is simply a temporary solution to the requirement of passing arguments of complex types to host functions. In the future, WasmEdge Runtime will support the component model spec that is the final solution to the issue. So, the current VmDock will be deprecated when the support comes. Thanks!

**Comment by yorkie** [comment]
Ok, I got it @apepkuss. 

However if the comment can be corrected, it will give users of `VmDock` a clearer understanding of its capabilities. I copied a copy of VmDock's code in my project and supported `named_module` way to call `run_func`, if you are willing to accept this patch, I am happy to submit a pull request (although `VmDock` will be deprecated).

In addition, I am also curious about what the Dock in the naming of `VmDock` specifically represents?

> Anyway, I'm closing this, thanks for you explaination @apepkuss
